flow/DeterministicRandom.cpp

```diff
@@ -93,11 +93,12 @@ uint64_t DeterministicRandom::randomUInt64() {
 }
 
 uint32_t DeterministicRandom::randomSkewedUInt32(uint32_t min, uint32_t maxPlusOne) {
-	std::uniform_real_distribution<double> distribution(std::log(min), std::log(maxPlusOne - 1));
-	double logpower = distribution(random);
-	uint32_t loguniform = static_cast<uint32_t>(std::pow(10, logpower));
-	// doubles can be imprecise, so let's make sure we don't violate an edge case.
-	return std::max(std::min(loguniform, maxPlusOne - 1), min);
+	ASSERT(min < maxPlusOne);
+	std::uniform_real_distribution<double> distribution(
+	    std::log(std::max<double>(min, 1.0 / M_E), std::log(maxPlusOne)));
```

Comment:
I think there's a misplaced parenthesis 

```suggestion
	    std::log(std::max<double>(min, 1.0 / M_E)), std::log(maxPlusOne));
```